tests/create_drillhole_data_test.py

```diff
@@ -47,43 +47,73 @@ def test_create_drillhole_data():
             ],
             name=well_name,
         )
-        distance = np.sort(np.random.rand(n_data) * max_depth)
         value_map = {}
         for ref in range(8):
             value_map[ref] = "".join(
                 random.choice(string.ascii_lowercase) for i in range(8)
             )
 
+        # Create random from-to
+        from_to_a = np.sort(
+            np.random.uniform(low=0.0, high=max_depth, size=(50,))
+        ).reshape((-1, 2))
+        from_to_b = np.vstack([from_to_a[0, :], [30.1, 55.5], [56.5, 80.2]])
+
         # Add from-to data
         data_objects = well.add_data(
             {
                 "interval_values": {
-                    "values": np.random.randn(n_data - 1),
-                    "from-to": np.c_[distance[:-1], distance[1:]],
-                },
-                "log_values": {
-                    "depth": distance,
-                    "type": "referenced",
-                    "values": np.random.randint(1, high=8, size=n_data),
-                    "value_map": value_map,
+                    "values": np.random.randn(from_to_a.shape[0]),
+                    "from-to": from_to_a,
                 },
                 "int_interval_list": {
                     "values": [1, 2, 3],
-                    "from-to": np.vstack([[0.25, 25.5], [30.1, 55.5], [56.5, 80.2]]),
+                    "from-to": from_to_b,
                     "value_map": {1: "Unit_A", 2: "Unit_B", 3: "Unit_C"},
                     "type": "referenced",
                 },
             }
         )
+
+        assert well.n_cells == (
+            from_to_a.shape[0] + 2
+        ), "Error with number of cells on interval data creation."
+        assert well.n_vertices == (
+            from_to_a.size + 4
+        ), "Error with number of vertices on interval data creation."
+        assert not np.any(
+            np.isnan(well.get_data("FROM")[0].values)
+        ), "FROM values not fully set."
+        assert not np.any(
+            np.isnan(well.get_data("TO")[0].values)
+        ), "TO values not fully set."
```

Comment:
**question (non-blocking):** should it also assert on the expected data size for `FROM` and `TO`?

Reply:
Added